processing/src/main/java/io/druid/segment/ConstantDimensionSelector.java

```diff
@@ -29,19 +30,35 @@
 import io.druid.segment.historical.SingleValueHistoricalDimensionSelector;
 
 import javax.annotation.Nullable;
+import java.util.Objects;
 
-public class NullDimensionSelector implements SingleValueHistoricalDimensionSelector, IdLookup
+public class ConstantDimensionSelector implements SingleValueHistoricalDimensionSelector, IdLookup
```

Comment:
I suggest to keep NullDimensionSelector. I think it's not a big deal, however it could make a difference for performance somewhere (reading a field vs. constant null)

Reply:
I thought about it, but then thought it wasn't worth the additional code since it should only affect performance of `lookupId` and `lookupName`, both of which are already incredibly cheap in ConstantDimensionSelector relative to a "real" dimension selector. And I didn't think it was worth having an entire extra file just for a probably very slight optimization for columns that don't exist.

I can add it if you still think it's worth it, but that was my reasoning.

Reply:
FYI #4623 allows to reduce code duplication in `ColumnSelectorFactory`, `GenericColumn`, etc. implementations, by leaving just `makeColumnValueSelector()` instead of `makeLong/Float/Double/Object/ColumnSelector()`. This will likely require to introduce `NilColumnValueSelector`, returning 0 from getLong/Float/Double, `null` from `get()`, `Void` from `classOfObject()`, and also implementing `DimensionSelector` exactly as currently `NullDimensionSelector` does.

Reply:
Good to know, but what do you think the implications are for this patch?

Reply:
That `NullDimensionSelector` will be revived and removing it now will only break git history for class.

But leaving this to you and this ask is not blocking for this PR.

Reply:
In that case, I will add it back in this PR, since it sounds like it will come back eventually anyway.